gostream/media_utils.go

```diff
@@ -90,6 +90,9 @@ func WithMIMETypeHint(ctx context.Context, mimeType string) context.Context {
 // set, the default provided is used.
 func MIMETypeHint(ctx context.Context, defaultType string) string {
 	if val, ok := ctx.Value(contextValueMIMETypeHint).(string); ok {
+		if val == "" {
+			return defaultType
+		}
 		return val
 	}
 	return defaultType
```

Comment:
```suggestion
	val, ok := ctx.Value(contextValueMIMETypeHint).(string)
	if !ok || val == "" {
		return defaultType
	}
	return val
```
[opt] you could combine the cases where we return `defaultType`.